R/plotreg.R

```diff
@@ -1,394 +1,372 @@
+        upper.inner <- dataframe$co + (z.inner * dataframe$se)
+        z.outer <- qnorm(1 - ((1 - ci.level) / 2))
+        lower.outer <- dataframe$co - (z.outer * dataframe$se)
+        upper.outer <- dataframe$co + (z.outer * dataframe$se)
+        
     } else if (length(models[[i]]@se) == 0 && length(models[[i]]@pvalues) > 0) {
-      stop("Model has p-values but no SEs. SEs or CIs are required for plotting.")
+        stop("Model has p-values but no SEs. SEs or CIs are required for plotting.")
     } else {
-      co <- models[[i]]@coef
-      co.names <- models[[i]]@coef.names
-      co <- co[-remove.rows]
-      co.names <- co.names[-remove.rows]
-      lower.outer <- models[[i]]@ci.low
-      upper.outer <- models[[i]]@ci.up
-      lower.inner <- NULL
-      upper.inner <- NULL
-      lower.outer <- lower.outer[-remove.rows]
-      upper.outer <- upper.outer[-remove.rows]
-      
-      # sort according to reorder.coef argument
-      dataframe <- data.frame(co.names, co, lower.outer, upper.outer)
-      dataframe <- reorder(dataframe, reorder.coef)
-      co.names <- dataframe[, 1]
-      co <- dataframe[, 2]
-      lower.outer <- dataframe[, 3]
-      upper.outer <- dataframe[, 4]
-      
-      signif.outer <- TRUE
-      note <- "Bars denote CIs."
-      message(paste0("Model ", i, ": bars denote  ", ci.level, 
-          " confidence intervals."))
+        lower.outer <- dataframe$ci.low
+        upper.outer <- dataframe$ci.up
+        lower.inner <- rep(0, length(dataframe$co.names))
+        upper.inner <- rep(0, length(dataframe$co.names))
+    }
+    
+    if (length(lower.outer) > 0) {
+        dataframe <- cbind(dataframe, lower.outer)     
+    }
+    if (length(lower.inner) > 0) {
+        dataframe <- cbind(dataframe, lower.inner)     
+    }
+    if (length(upper.outer) > 0) {
+        dataframe <- cbind(dataframe, upper.outer)     
+    }    
+    if (length(upper.inner) > 0) {
+        dataframe <- cbind(dataframe, upper.inner)     
     }
+    if (length(dataframe$pv) > 0) {
+        signif.outer <- dataframe$pv < (1 - ci.level)
+    } 
     
-    if (!is.null(custom.note)) {
-      note <- custom.note
+    # which terms are significant?
+    if(length(dataframe$ci.low) > 0) {
+        signif.outer <- ((dataframe$ci.low) > 0 & (dataframe$ci.up) > 0 | (dataframe$ci.low) < 0 & (dataframe$ci.up) < 0)
     }
     
+    if (class(signif.outer) == "logical" &&
+        length(signif.outer) == length(dataframe$co)) {
+        signif <- signif.outer == FALSE
+    } else if (signif.outer == TRUE) {
+        signif <- apply(cbind(lower.outer, upper.outer), 1,
+                        function(x) x[1] <= 0 && x[2] >= 0)
+    } else if (signif.outer == FALSE) {
+        signif <- apply(cbind(lower.inner, upper.inner), 1,
+                        function(x) x[1] <= 0 && x[2] >= 0)
+    } else {
+        stop("signif.outer does not correspond to the intervals provided.")
```

Comment:
Put `signif.outer` in single quotes. Also: why do you have to throw an error? Since you are creating that object, you should know its properties, right? But maybe I'm overlooking something.

Reply:
this error was in the old plotreg (or coefplot) coding and I just left it there. If you think it is no longer necessary I'm happy to remove it.

Reply:
Ah okay. Let's leave it there. I sometimes don't remember what I wrote...